FaunaDB.Client/Client/FaunaClient.cs

```diff
@@ -27,12 +27,28 @@ public class FaunaClient
         /// <summary>
         /// Constructs a new instance of <see cref="FaunaClient"/> with the specified arguments.
         /// </summary>
+        /// <param name="secret">Auth token for the FaunaDB server.</param>
+        /// <param name="endpoint">URL for the FaunaDB server. Defaults to "https://db.fauna.com:443"</param>
+        /// <param name="timeout">Timeout for I/O operations. Defaults to 1 minute.</param>
+        /// <param name="apiVersion">FaunaDB API Version. Defaults to <see cref="ApiVersion.VERSION_2_0"/>.</param>
+        public FaunaClient(
+            string secret,
+            string endpoint = "https://db.fauna.com:443",
+            TimeSpan? timeout = null,
+            ApiVersion? apiVersion = null)
+            : this(CreateClient(secret, endpoint, timeout, apiVersion))
+        { }
+
+        /// <summary>
+        /// Constructs a new instance of <see cref="FaunaClient"/> with the specified arguments.
+        /// </summary>
         /// <param name="domain">Base URL for the FaunaDB server.</param>
         /// <param name="scheme">Scheme of the FaunaDB server. Should be "http" or "https". Default is "https"</param>
         /// <param name="port">Port of the FaunaDB server. Defaults to 443 if schema is "https" or 80 if schema is "http"</param>
         /// <param name="timeout">Timeout for I/O operations. Defaults to 1 minute.</param>
         /// <param name="secret">Auth token for the FaunaDB server.</param>
         /// <param name="clientIO">Optional <see cref="IClientIO"/>.</param>
+        [Obsolete("prefer FaunaClient(secret:endpoint:timeout:apiVersion:) instead")]
```

Comment:
The reason is because in my option this constructor was bad designed. The last argument is exclusive, I mean, if you pass in a `ClientIO` it will ignore the others. That's way I also didn't retrofit it with api version.